play-media-native/screens/EditExistingMedia.tsx

```diff
@@ -0,0 +1,255 @@
+import { Screen } from '../features/Screen/Screen';
+import { removeFileExtension } from '../helpers/media';
+import { useMediaQuery } from '../hooks/useMediaQuery/useMediaQuery';
+import { Media } from '../interfaces/media';
+import { RootStackParamList } from '../interfaces/navigators';
+import { styles } from '../theme/styles';
+import { theme } from '../theme/theme';
+
+type Props = NativeStackScreenProps<RootStackParamList, 'EditExistingMedia'>;
+
+export const EditExistingMediaScreen = ({ navigation, route }: Props) => {
+  const id = route?.params?.id;
+
+  const [editedMedia, setEditedMedia] = useState<Partial<Media>>();
+  const [isMediaItemSaved, setIsMediaItemSaved] = useState(false);
+  const [isValidating, setIsValidating] = useState(false);
+  const [isPublished, setIsPublished] = useState(false);
+  const [shouldShowSuccessView, setShouldShowSuccessView] = useState(false);
+  const [shouldShowErrorView, setShouldShowErrorView] = useState(false);
+
+  const { error, isFetching } = useQuery(`media-${id}`, () => getMediaById(id), {
+    staleTime: 0,
+    onSuccess: (data) => setEditedMedia(data),
+  });
+
+  // In case of publishing we have to manually update the media item's status
+  // because the server takes too long to reflect the change
+  const [mediaID, setMediaID] = useState(null);
+  const [mediaStatus, setMediaStatus] = useState(null);
+  const { refetch: refetchMediaListing } = useMediaQuery(mediaID, mediaStatus);
+
+  const headerTitle = useMemo(
+    () => (editedMedia?.name ? removeFileExtension(editedMedia.name) : 'Untitled media'),
+    [editedMedia?.name]
+  );
+  const editedMediaTitle = useMemo(
+    () => (editedMedia?.name ? removeFileExtension(editedMedia.name) : ''),
+    [editedMedia?.name]
+  );
+
+  useFocusEffect(
+    useCallback(() => {
+      navigation.setParams({
+        title: headerTitle,
+      });
+    }, [headerTitle, navigation])
+  );
+
+  useFocusEffect(
+    useCallback(() => {
+      const unsubscribe = navigation.addListener('beforeRemove', (event) => {
+        if (!isMediaItemSaved) {
+          event.preventDefault();
+
+          navigation.push('DiscardChanges', {
+            id: editedMedia?.id,
+            title: headerTitle,
+            subtitle: 'Discard media file changes?',
+            message: 'Are you sure you want to discard the media file changes?',
+            redirectRoute: 'MediaDetail',
+          });
+        }
+      });
+
+      return () => {
+        unsubscribe();
+      };
+    }, [navigation, isMediaItemSaved, editedMedia?.id, headerTitle])
+  );
+
+  const handleTitleChange = useCallback((text: string) => {
+    setEditedMedia((media) => ({
+      ...media,
+      name: text,
+    }));
+  }, []);
+
+  const handleDescriptionChange = useCallback((text: string) => {
+    setEditedMedia((media) => ({
+      ...media,
+      description: text,
+    }));
+  }, []);
+
+  const processSuccess = useCallback(
+    async (id = undefined, status = undefined) => {
```

Comment:
ID and status are not really being used in the function apart from being a condition - maybe it could be a boolean that's being passed.